### PR TITLE
utils: add support to package the Windows toolchain

### PR DESCRIPTION
**Pull request by compnerd**
Package up the toolchain into the MSI set for distribution.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @shahmishal

File: utils/build-windows-toolchain.bat [hunk 1]
```diff
@@ -585,6 +595,73 @@ python -c "import plistlib; print(str(plistlib.dumps({ 'DefaultProperties': { 'D
...
+move %PackageRoot%\runtime\runtime.msi %PackageRoot% || (exit /b)
+move %PackageRoot%\icu\icu.msi %PackageRoot% || (exit /b)
+move %PackageRoot%\devtools\devtools.msi %PackageRoot% || (exit /b)
+
+:: Build Installer
+msbuild %SourceRoot%\swift-installer-scripts\platforms\Windows\installer.wixproj ^
+  -p:RunWixToolsOutOfProc=true ^
+  -p:OutputPath=%PackageRoot%\installer\ ^
+  -p:IntermediateOutputPath=%PackageRoot%\installer\ ^
+  -p:MSI_LOCATION=%PackageRoot%\
```
Comment (shahmishal) [inline utils/build-windows-toolchain.bat:661]: Do we need the `\` here?

File: utils/build-windows-toolchain.bat [hunk 2]
```diff
@@ -585,6 +595,73 @@ python -c "import plistlib; print(str(plistlib.dumps({ 'DefaultProperties': { 'D
...
+  -p:PlatformToolset=v142
+:: TODO(compnerd) actually perform the code-signing
+:: signtool sign /f Apple_CodeSign.pfx /p Apple_CodeSign_Password /tr http://timestamp.digicert.com /fd sha256 %PackageRoot%\sdk\sdk.msi
+
+:: Package runtime.msi
+msbuild %SourceRoot%\swift-installer-scripts\platforms\Windows\runtime.wixproj ^
+  -p:RunWixToolsOutOfProc=true ^
+  -p:OutputPath=%PackageRoot%\runtime\ ^
+  -p:IntermediateOutputPath=%PackageRoot%\runtime\ ^
+  -p:SDK_ROOT=%SDKInstallRoot%\
```
Comment (shahmishal) [inline utils/build-windows-toolchain.bat:625]: Also, here too (do we need `\`?)

File: utils/build-windows-toolchain.bat (hunk 1 above)
Comment (compnerd) [reply utils/build-windows-toolchain.bat:661]: Yes, in general, its far safer to provide the trailing `\` for paths.

File: utils/build-windows-toolchain.bat (hunk 2 above)
Comment (compnerd) [reply utils/build-windows-toolchain.bat:625]: Similar.

**Review by shahmishal — APPROVED** [review]

**Pull request merged**